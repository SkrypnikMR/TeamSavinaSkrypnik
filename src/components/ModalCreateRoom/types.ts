export type TModalCreateRoom = {
    handlecloseModal:() => void;
    createRoom: (arg: string) => void;
}
