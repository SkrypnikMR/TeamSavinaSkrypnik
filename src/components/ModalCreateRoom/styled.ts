import styled from 'styled-components';

// import { lightGray } from '../UI/baseLayout';

export const StModalButtonBox = styled.div`
  margin: 10px;
  display: flex;
  justify-content: space-between;
`;
