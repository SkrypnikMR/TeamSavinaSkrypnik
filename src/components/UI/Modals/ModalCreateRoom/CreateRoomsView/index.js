import { connect } from 'react-redux';
import { changeModalVisibility } from 'src/Store/modals/actions';
import CreateRoomsView from './CreateRoomsView.jsx';

const mapDispatchToProps = dispatch => ({
    changeModalVisibility: payload => dispatch(changeModalVisibility(payload)),
});

export default connect(null, mapDispatchToProps)(CreateRoomsView);
