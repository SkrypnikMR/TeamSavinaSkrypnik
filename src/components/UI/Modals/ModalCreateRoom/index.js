import CreateUsesrView from './ModalCreateRoom.jsx';

export default CreateUsesrView;
