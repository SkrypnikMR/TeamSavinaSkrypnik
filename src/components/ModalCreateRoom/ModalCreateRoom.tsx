import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import Portal from '../Portal';
import Button from '../UI/Button';
import { GAME_TYPE } from '../../constants/componentsÐ¡onsts';
import { lightGray } from '../UI/baseLayout';

import { StModalCreateRoom, StModalContent, StModalContentItems, StModalHeader, StModalButtonBox } from './styled';
import Select from '../UI/Select/Select';

const ModalCreateRoom = ({ handlecloseModal, createRoom }) => {
    const [state, setState] = useState({ gameType: GAME_TYPE.tic_tac_toe });
    const { t } = useTranslation();
    const handleModalClick = () => {
        createRoom(state.gameType);
        handlecloseModal();
    };
    const handleOnchange = (e) => setState({ ...state, gameType: e.target.value });
    return (
        <Portal>
            <StModalCreateRoom>
                <StModalContent>
                    <Button 
                        content="X"
                        onClick={handlecloseModal}
                        id="CloseButton"
                        name="CloseButton"
                        value="Close"
                        type="button"
                        title="Close"
                        height="40px"
                        width="40px"
                        bgColor={lightGray}
                        position="absolute"
                        right="10px"
                    />
                    <StModalContentItems>
                        <StModalHeader>
                            {t('create_new_game')}
                        </StModalHeader>
                        {/* <select onChange={handleOnchange} value={state.gameType}>
                            <option value={GAME_TYPE.tic_tac_toe}>{t('tic_tac_toe')}</option>
                            <option value={GAME_TYPE.checkers}>{t('Checkers')}</option>
                        </select> */}
                        <Select onChange={handleOnchange} value={state.gameType} options={[GAME_TYPE.tic_tac_toe, GAME_TYPE.checkers]}/>
                        <StModalButtonBox>
                            <Button 
                                content="OK"
                                focusColor="black"
                                onClick={handleModalClick}
                                id="okButton"
                                name="okButton"
                                value="OK"
                                type="button"
                                title="OK"
                            />
                            <Button
                                content={t('cancel')}
                                focusColor="black"
                                onClick={handlecloseModal}
                                id="CancelButton"
                                name="CancelButton"
                                value="Cancel"
                                type="button"
                                title="Cancel"
                            />
                        </StModalButtonBox>
                    </StModalContentItems>
                </StModalContent>
            </StModalCreateRoom>
        </Portal>
    );
};

export default ModalCreateRoom;
