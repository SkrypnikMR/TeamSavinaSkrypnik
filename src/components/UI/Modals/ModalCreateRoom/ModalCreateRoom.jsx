import * as React from 'react';
import ModalComponent from '../../Modal';
import CreateRoomsView from './CreateRoomsView';

const ModalCreateRoom = () => {
    const modalType = 'createChat';
    return (
        <ModalComponent
            headerTextKey='create_chat'
            modalType={modalType}
        >
            <CreateRoomsView modalType={modalType} />
        </ModalComponent>
    );
};

export default ModalCreateRoom;
