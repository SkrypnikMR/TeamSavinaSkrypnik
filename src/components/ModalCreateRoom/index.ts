import ModalCreateRoom from './ModalCreateRoom';

export default ModalCreateRoom;
