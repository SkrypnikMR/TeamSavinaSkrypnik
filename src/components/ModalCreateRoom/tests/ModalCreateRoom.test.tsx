import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import ModalCreateRoom from '../ModalCreateRoom';
import { act } from 'react-dom/test-utils';

const element = document.createElement('div');
element.setAttribute('id', 'modal');
element.setAttribute('data-testid', 'modal-test-id');

jest
    .spyOn(ReactDOM, 'createPortal')
    .mockImplementation((children, c, key) => {
        const symbol = Symbol.for('react.portal');
        return {
            $$typeof: symbol,
            key: key == null ? null : '' + key,
            children,
            containerInfo: element,
            implementation: null,
            type: symbol.description,
            props: null,
        } as ReactPortal;
    });

describe('ModalCreateRoom', () => {
    let props;
    beforeEach(() => {
    props = {
        handlecloseModal: jest.fn(),
        createRoom: jest.fn(),
    };
        });
    it('should render StModalCreateRoom', () => {
        const component = mount(<ModalCreateRoom {...props}/>);
        expect(component.find('styled__StModalCreateRoom')).toHaveLength(1);
    });
    it('should click on first button, then call createRoom and handlecloseModal', ()=>{
        const component = mount(<ModalCreateRoom {...props}/>);
        component.find('Button').at(0).props().onClick();
        expect(props.createRoom).toHaveBeenCalled();
        expect(props.handlecloseModal).toHaveBeenCalled();
    })
    it('should click on second button, then call only handleCloseModal', ()=>{
        const component = mount(<ModalCreateRoom {...props}/>);
        component.find('Button').at(1).props().onClick();
        expect(props.handlecloseModal).toHaveBeenCalled();
    })
    it('should trigger onChange', ()=>{
        const testValue = 'testvValue';
        const component = mount(<ModalCreateRoom {...props}/>);
        act(()=>{
            component.find('select').props().onChange({target: {value: testValue}});
        })
        component.update();
        expect(component.find('select').props().value).toBe(testValue);
    })
});
