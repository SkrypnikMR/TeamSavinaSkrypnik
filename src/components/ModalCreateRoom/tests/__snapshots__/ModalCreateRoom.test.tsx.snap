// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCreateRoom should snapshot 1`] = `"<div><select class=\\"styled__StSelect-sc-1i888oy-0 ijdFJw\\"><option selected=\\"\\" label=\\"Tic-tac-toe\\">Tic-tac-toe</option><option label=\\"Checkers\\">Checkers</option></select><div class=\\"styled__StModalButtonBox-sc-1lbp0ln-0 kgpYEL\\"><button id=\\"okButton\\" name=\\"okButton\\" type=\\"button\\" value=\\"OK\\" title=\\"OK\\" class=\\"styled__StButton-sc-1y35zax-0 lkPeFg\\">OK</button><button id=\\"CancelButton\\" name=\\"CancelButton\\" type=\\"button\\" value=\\"Cancel\\" title=\\"Cancel\\" class=\\"styled__StButton-sc-1y35zax-0 efYwVL\\">cancel</button></div></div>"`;
